tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -15320,17 +15421,23 @@ void TryAddShaderAttrFromTargetProfile(Sema &S, FunctionDecl *FD) {
       HLSLShaderAttr::CreateImplicit(S.Context, fullName);
 
   FD->addAttr(pShaderAttr);
+  isActiveEntry = true;
   return;
 }
 
 void DiagnoseEntry(Sema &S, FunctionDecl *FD) {
-  TryAddShaderAttrFromTargetProfile(S, FD);
+  bool isActiveEntry = false;
+  TryAddShaderAttrFromTargetProfile(S, FD, isActiveEntry);
 
-  auto Attr = FD->getAttr<HLSLShaderAttr>();
+  HLSLShaderAttr *Attr = FD->getAttr<HLSLShaderAttr>();
   if (!Attr) {
     return;
   }
 
+  if (isActiveEntry) {
```

Comment:
This should be on inactive entries as well, so no need for this `if (isActiveEntry)` here.